### PR TITLE
DAH 2142 salesforce cleanup job

### PR DESCRIPTION
**Pull request by tallulahkay**
DAH-2142

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -183,3 +203,10 @@ workflows:
           name: e2e-production
           requires:
             - build
+      - waiter:
```
Comment (tallulahkay) [inline .circleci/config.yml]: We need to make sure it also runs on prod

**Review by alulabeshue-sfgov — COMMENTED** [review]
question: In this partners PR you are running the cleanup job without having to run the rails server, just calling a rake job. In [webapp](https://github.com/SFDigitalServices/sf-dahlia-web/blob/main/.circleci/config.yml#L249) we are running the rails server before calling the rake job. Do we need the additional steps in webapp to start the rails server? The start server step in circleci adds some time to the build so was wondering if we need it.

**Comment by tallulahkay** [comment]
@alulabeshue-sfgov you're so right - don't think we need the start-server part in webapp - I'll make a ticket to confirm and remove

**Review by alulabeshue-sfgov — APPROVED** [review]
LGTM, thanks!

**Comment by tallulahkay** [comment]
I converted to draft bc reggie is changing the endpoint to a DELETE

**Pull request merged**